server/tserver/src/main/java/org/apache/accumulo/tserver/compaction/EverythingCompactionStrategy.java

```diff
@@ -27,7 +27,7 @@
 
   @Override
   public boolean shouldCompact(MajorCompactionRequest request) throws IOException {
-    return request.getFiles().size() > 0;
+      return true; // ACCUMULO-3645 compact for empty files too
```

Comment:
A point to consider: unless it's definitely a bug fix, and we're sure we want to change the behavior to fix that bug, our versioning rules would disallow new functionality to be added in 1.6.3. It'd have to go in 1.7.